apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1022,11 +1022,29 @@ func (m *ModelManagerAPI) DestroyModels(ctx context.Context, args params.Destroy
 		// TODO (tlm): The modelService nil check will go when the tests are
 		// moved from mongo.
 		if m.modelService != nil {
-			err = m.modelService.DeleteModel(ctx, coremodel.UUID(stModel.UUID()))
+			// We need to get the model service factory from the model
+			// We should be able to directly access the model service factory
+			// because the model manager use the MultiModelContext to access
+			// other models.
+			modelUUID := coremodel.UUID(stModel.UUID())
+
+			// TODO (stickupkid): We need to delete the model info when
```

Comment:
This is because we need to implement the life cycle for a model. We're currently going from `Alive -> Dead` in the model manager, instead of `Alive -> Dying -> Dead`